ocaml-lsp-server/test/e2e-new/type_enclosing.ml

```diff
@@ -0,0 +1,431 @@
+open Test.Import
+
+let call_type_enclosing ?(verbosity = 0) ?range_end client position index =
+  let uri = DocumentUri.of_path "test.ml" in
+  let text_document = TextDocumentIdentifier.create ~uri in
+  let params =
+    `Assoc
+      ([ ("textDocument", TextDocumentIdentifier.yojson_of_t text_document)
+       ; ("position", Position.yojson_of_t position)
+       ; ("index", `Int index)
+       ; ("verbosity", `Int verbosity)
+       ]
+      @
+      match range_end with
+      | None -> []
+      | Some x -> [ ("rangeEnd", Position.yojson_of_t x) ])
+  in
+  let params = Some (Jsonrpc.Structured.t_of_yojson params) in
+  let req =
+    Lsp.Client_request.UnknownRequest
+      { meth = "ocamllsp/typeEnclosing"; params }
+  in
+  Client.request client req
+
+let print_type_enclosing result =
+  result |> Yojson.Safe.pretty_to_string ~std:false |> print_endline
+
+let%expect_test "type enclosing on simple example" =
+  let source = {|let x = string_of_int 2002|} in
+  let request client =
+    let open Fiber.O in
+    let position = Position.create ~line:0 ~character:22 in
```

Comment:
I am trying to read the tests: these are positions 0-based right ? For the line it looks like it is the case, and for the column (`character`) number too ?